### PR TITLE
feat: correlate vulnerabilties for purls using product statuses

### PR DESCRIPTION
**Pull request by dejanb**
This PR addresses #860

### CODE REVIEW

File: modules/fundamental/src/purl/model/details/purl.rs [hunk 1]
```diff
@@ -275,3 +395,33 @@ impl FromQueryResultMultiModel for LicenseCatcher {
             .try_model_columns(license::Entity)
     }
 }
+
+#[derive(Debug)]
+#[allow(dead_code)] //TODO sbom field is not used at the moment, but we will probably need it for graph search
```
Comment (ctron) [inline modules/fundamental/src/purl/model/details/purl.rs:400]: Does this comment still apply? It says "field".

Comment (dejanb) [reply modules/fundamental/src/purl/model/details/purl.rs]: It doesn't. It was copied over from another query catcher. Fixed now, thanks for spotting it.

**Review by ctron — COMMENTED** [review]
:+1:  LGTM. But I'd want @JimFuller-RedHat to have a look too, especially at the DB related stuff.

**Review by JimFuller-RedHat — APPROVED** [review]
I think this is fine to get into master - it will need some optimisation but I will do that as part of final perf pass, LGTM

**Pull request merged**